src/core/loader/cia.h

```diff
@@ -0,0 +1,367 @@
+//
+// Created by Jfm Meyers on 3/15/17.
+//
+#pragma once
+#include <string>
+#include "common/common_types.h"
+#include "core/loader/loader.h"
+#include <common/common_paths.h>
+
+namespace Loader {
+/// Loads an CIA file
+// File header
+typedef struct
+{
+  u8 headersize[4];
+  u8 type[2];
+  u8 version[2];
+  u8 certsize[4];
+  u8 ticketsize[4];
+  u8 tmdsize[4];
+  u8 metasize[4];
+  u8 contentsize[8];
+  u8 contentindex[0x2000];
+} ctr_ciaheader;
+
+#define MAGIC_NCCH 0x4843434E
+#define MAGIC_NCSD 0x4453434E
+#define MAGIC_FIRM 0x4D524946
+#define MAGIC_CWAV 0x56415743
+#define MAGIC_IVFC 0x43465649
+
+#define SIZE_128MB (128 * 1024 * 1024)
+
+
+
+typedef unsigned long t_int;
+typedef struct
+{
+  int s;              /*!<  integer sign      */
+  int n;              /*!<  total # of limbs  */
+  t_int *p;           /*!<  pointer to limbs  */
+}
+    mpi;
```

Comment:
what happened here? This should probably be on the same line [also see my other comment about the typedefs]